### PR TITLE
Fixed port of mongo-c-driver/mongo-cxx-driver for android

### PR DESCRIPTION
**Pull request by Markus87**
This fixes building of mongo-c-driver and mongo-cxx-driver for android.

Since android is still unofficial I tested it with the triplets defined here: https://vcpkg.readthedocs.io/en/latest/users/android/
I tested `arm-android` and `arm64-android` with `VCPKG_LIBRARY_LINKAGE = static`

### CODE REVIEW

**Review by JackBoosY — COMMENTED** [review]
Please bump the port version. See [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).

File: ports/mongo-c-driver/portfile.cmake [hunk 1]
```diff
@@ -28,6 +28,14 @@ else()
     set(ENABLE_SSL "OPENSSL")
 endif()
 
+if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Android")
```
Comment (JackBoosY) [inline ports/mongo-c-driver/portfile.cmake:31]:
```suggestion
if(VCPKG_TARGET_IS_ANDROID)
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
Forgot update `Port-Version`?

**Comment by Markus87** [comment]
@JackBoosY No, I applied your suggestion in github but made the changes to the manifest files later.

**Review by JackBoosY — APPROVED** [review]

**Comment by dan-shaw** [comment]
Thanks!

**Pull request merged**